Apps/Encounter/src/Delegates/RestHospitalVisitDelegate.cs

```diff
@@ -91,71 +92,25 @@ public async Task<RequestResult<PhsaResult<IEnumerable<HospitalVisit>>>> GetHosp
 
             try
             {
-                IApiResponse<PhsaResult<IEnumerable<HospitalVisit>>> response =
-                    await this.hospitalVisitApi.GetHospitalVisits(query, accessToken).ConfigureAwait(true);
-                this.ProcessResponse(requestResult, response);
+                PhsaResult<IEnumerable<HospitalVisit>> response =
+                    await this.hospitalVisitApi.GetHospitalVisitsAsync(query, accessToken).ConfigureAwait(true);
+                requestResult.ResultStatus = ResultType.Success;
+                requestResult.ResourcePayload!.Result = response.Result;
+                requestResult.TotalResultCount = requestResult.ResourcePayload.Result.Count();
             }
-            catch (HttpRequestException e)
+            catch (Exception e) when (e is ApiException or HttpRequestException)
             {
-                this.logger.LogCritical("HTTP Request Exception {Error}", e.ToString());
+                this.logger.LogError("Error while retrieving Hospital Visits... {Error}", e);
+                HttpStatusCode? statusCode = (e as ApiException)?.StatusCode ?? ((HttpRequestException)e).StatusCode;
                 requestResult.ResultError = new()
                 {
-                    ResultMessage = "Error with HTTP Request",
+                    ResultMessage = $"Status: {statusCode}. Error while retrieving Hospital Visits",
                     ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.Phsa),
                 };
```

Comment:
We should discuss if we want to include specific information (e.g. status code) in the the result message moving forward.

Reply:
I don't think we should

Reply:
Okay I can change that. I was using the example from RestMspVisitDelegate in Encounter, should I change it there too?

Reply:
The message should at least differentiate the ApiException from the HttpRequestException.  Having a code or a more definitive message is better than a generic one where you will be forced to search through Kibana for why this occurred.

Moreover, these exceptions were rarely occur.  These are not validation errors.

Reply:
Also the status code is retrieved from the ApiException and HttpRequestException.  No problems with a generic message.